karibu-dsl/src/main/kotlin/com/github/mvysny/karibudsl/v10/Router.kt

```diff
@@ -2,39 +2,39 @@ package com.github.mvysny.karibudsl.v10
 
 import com.vaadin.flow.component.Component
 import com.vaadin.flow.component.HasComponents
-import com.vaadin.flow.component.icon.VaadinIcon
+import com.vaadin.flow.component.icon.IconFactory
 import com.vaadin.flow.router.HasUrlParameter
 import com.vaadin.flow.router.RouterLink
 import kotlin.reflect.KClass
 import com.vaadin.flow.router.RouteParameters
 
 @VaadinDsl
-public fun (@VaadinDsl HasComponents).routerLink(icon: VaadinIcon? = null, text: String? = null, viewType: KClass<out Component>,
+public fun (@VaadinDsl HasComponents).routerLink(icon: IconFactory? = null, text: String? = null, viewType: KClass<out Component>,
                                                  block: (@VaadinDsl RouterLink).() -> Unit = {}): RouterLink {
     val link = RouterLink(null as String?, viewType.java)
-    if (icon != null) link.icon(icon)
+    if (icon != null) link.init(icon.create())
```

Comment:
Hi, I've updated the `icon{}` DSL to also accept `IconFactory`. It should work as-is, could you please update & revert this line?

Reply:
Hi! I did it.